driver/network_driver.go

```diff
@@ -286,12 +286,12 @@ func (d NetworkDriver) CreateEndpoint(request *network.CreateEndpointRequest) (*
 		// the request if the profile already exists.
 		profile := &api.Profile{
 			Metadata: api.ProfileMetadata{
-				Name: networkData.Name,
-				Tags: []string{networkData.Name},
+				Name:   networkData.Name,
+				Labels: map[string]string{"projectcalico.org/group": networkData.Name},
```

Comment:
For the best back-compat story, I think we should do `Labels: map[string]string{networkData.Name:""}`

That will be equivalent to the transform that felix is doing to the tags.  In fact, it'll make sure that old matches on tags keep working because we now map a match of tag `foo` to a selector `has(foo)`.